src/reorders/reorderCatArgs.ts

```diff
@@ -17,6 +17,8 @@ export default (function reorderCatArgs(
       catArgsManager.makeNewCatArrs(store, inst_axis, cat_index);
       // update the attribute
       catArgsManager.updateCatArgsAttribute(inst_axis, cat_index);
+
+      catArgsManager.regenerateCatArgsArrs(store);
     }
```

Comment:
Zooming in and out  of the matrix removed the current state of the categories and it was setting it back to the initial, so have added this to prevent that.